api/transport/handler.go

```diff
@@ -168,13 +169,30 @@ func DispatchOnewayHandler(
 func DispatchStreamHandler(
 	h StreamHandler,
 	stream *ServerStream,
+	logger *zap.Logger,
 ) (err error) {
 	defer func() {
 		if r := recover(); r != nil {
-			log.Printf("Stream handler panicked: %v\n%s", r, debug.Stack())
-			err = fmt.Errorf("panic: %v", r)
+			err = logPanic("Stream", logger, r, stream.Request().Meta)
 		}
 	}()
 
 	return h.HandleStream(stream)
 }
+
+func logPanic(rpcType string, logger *zap.Logger, recovered interface{}, req *RequestMeta) error {
```

Comment:
does it make sense to use the transport.Type enum to map a string to log here? package level?

Reply:
`transport.Type` implements Stringer so that can be passed here as-is and we can call `.String()` on it -- or use `%v` in the `fmt.Sprintf` where it's used (which calls `String()` on the object if it has one).

Reply:
Good intuition @zmt! My spidey sense was def going off here, but I follow the [make it, make it work, make it right] mantra. Let's update this now!

Reply:
The call sites in Dispatch* do not have the HandlerSpec with the type in scope, so wiring that in there would be a breaking change in the API, so going to leave the hard-coded strings here.

Reply:
thanks @abg - it was simpler than I was imagining.